internal/dataplane/kongstate/kongstate.go

```diff
@@ -77,7 +83,10 @@ func (ks *KongState) FillConsumersAndCredentials(log logrus.FieldLogger, s store
 			})
 			secret, err := s.GetSecret(consumer.Namespace, cred)
 			if err != nil {
-				log.WithError(err).Error("failed to fetch secret")
+				failureCollector.PushResourceFailure(
+					fmt.Sprintf("failed to fetch secret: %v", err),
+					consumer,
+				)
```

Comment:
I think it would be useful to provide context on what credential every failure is reported for. In error logs, we had `secret_name` and `secret_namespace` set, maybe we could prepare a closure in every `for _, cred := range consumer.Credentials {` run, something like this:

```go
pushCredResourceFailures := func(message string) {
    failuresCollector.PushResourceFailure(fmt.Sprintf("credential %q failure: %s", cred, message), consumer)
}
```